simulation/executor/types.go

```diff
@@ -0,0 +1,17 @@
+package simulation
+
+import (
+	"encoding/json"
+	"math/rand"
+	"time"
+
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+)
+
+// AppStateFn returns the app state json bytes and the genesis accounts
+type AppStateFn func(r *rand.Rand, accs []simtypes.Account, config Config) (
+	appState json.RawMessage, accounts []simtypes.Account, chainId string, genesisTimestamp time.Time,
+)
+
+// RandomAccountFn returns a slice of n random simulation accounts
+type RandomAccountFn func(r *rand.Rand, n int) []simtypes.Account
```

Comment:
I'm confused why were adding these types?

AppStateFn is a leagcy type, that new things shouldn't be using. We just have references to it, for the legacy actions / genesis handling

Reply:
Without calling these from osmosis, we are forced to call them from simulation (sdk side) which uses a simulation.Config type rather than a Config type. We want to use our local definition of Config since this is what holds all the flags etc. Please lmk if I am misunderstanding

Reply:
hrmm, not sure I understand still, but this isn't a big deal / we can just change it later